src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -68,6 +68,12 @@ def list(self):
             osd['stats'] = {}
             osd['stats_history'] = {}
             osd_spec = str(osd_id)
+            metadata = mgr.get_metadata('osd', osd_spec)
```

Comment:
Could you please measure [how long this `list()` method takes](https://stackoverflow.com/a/27737385/13173750) with: 1, 2, 4, 8 and 16 OSDs. You can easily do that in vstart (`OSD=N` env var setting). It's also be interesting to see how the CPU reacts during this request (you'll probably better launch this from REST via curl CLI).

Reply:
With the data that @aaSharma14 obtained from running this with OSD count {1,2,4,6,8,16}, as expected, it grows linearly (not many samples but enough to confirm it). Despite not resulting in huge numbers when extrapolated to OSD=100 or even 1000 (35 ms or 350 ms), the number of API calls (O(N)) might cause issues with the processing queues at ceph-mgr.

![image](https://user-images.githubusercontent.com/37327689/97331598-dba40100-1879-11eb-966b-fd3c0d07bb6f.png)

Additionally, this `mgr.get_metadata()` call is only used for retrieved the number of devices of each OSD, which is a rarely changing data.

Solutions:
* To find a single ceph-mgr API call returning this info aggregated for all OSDs.
* Or to introduce pagination, or at least only retrieving separately the currently displayed OSD information, and the device count (only for the OSDs displayed). As a side effect of this, the Device Count column couldn't be sortable/searchable anymore (we don't have all OSD's device count to sort that or search).

Reply:
This PR meant to fix 2 issues reported in the linked tracker (@pcuzner please check this):

> 1. The table provides a list of OSDs and yet the first column isn’t the osd id it’s the host
>   1.1. Put the osd id first
> 2. Each osd can span multiple devices (and does for spinners), but to understand how many devices support an OSD you have to go to the properties of each OSD..
>   2.1. Provide the count of devices an OSD uses on each row to provide a quick way for the configuration to be checked.

The 1st one is tackled, while the second 2 is "feasible" but would impact scalability: OSD endpoint is now O(1) (as long as `mgr.get('osd_map')` is O(1) which is probably an 'over-assumption' given CPython interfacing overhead) and for retrieving this information, it's necessary to poll `mgr.metadata()` for every OSD, which is O(N) for a periodically polled endpoint...

My suggestion is to keep the current behaviour: leave the device count in the OSD Details sub-table, and only poll for that information when a user wants to expand details for that OSD. Additionally, the device count is 'low-entropy' data (will never change once an OSD is created, and most, if no all, OSDs will have the same device count).

Reply:
Thanks @epuertat . I have removed the device count feature as discussed and this PR is now only intended for :
The table provides a list of OSDs and yet the first column isn’t the osd id it’s the host
1.1. Put the osd id first

Reply:
@aaSharma14 Could you update the description with a screenshot?